src/client/LoginForm.jsx

```diff
@@ -59,6 +12,44 @@ const LoginForm = ({close, setUser, setLoggedIn, siteKey, isShowRememberMe = tru
     const [password, setPassword] = useState('');
     const [rememberMe, setRememberMe] = useState(false);
 
+    const login = () => {
+        const body = [
+            'username=' + username,
+            'password=' + password
+        ];
+
+        if (rememberMe) {
+            body.push('useCookie=on');
+        }
+
+        fetch(loginUrl + '?restMode=true' + (siteKey ? `&site=${siteKey}` : ''), {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+                'allow-redirects': 'false'
+            },
+            body: body.join('&')
+        }).then(response => {
+            try {
+                response.body.getReader().read().then(({value}) => {
+                    const decodedValue = new TextDecoder().decode(value);
+                    if (decodedValue === 'OK') {
+                        close();
+                        setUser(username);
+                        setLoggedIn(true);
+                    } else if (decodedValue === 'unauthorized') {
+                        setIncorrectLogin(true);
+                    } else {
+                        throw new Error();
+                    }
+                });
+            } catch (e) {
+                console.error('Login form error : ', e);
+                setUnknownError(true);
+            }
```

Comment:
This `catch` block cannot catch anything because of the asynchronous nature of this function call.

It would be best to rewrite as such (you'll need to mark `login` as `async`):

```js
try {
  const response = await fetch(...);
  const value = await response.text(); // better than getReader().read()
  if (...) { ... }
} catch (error) {
  ...
}
```

If you want to preserve the chain of callbacks, you can write it as such:

```js
fetch(...)
	.then(response => response.text())
	.then(value => {
		...
	})
	.catch(error => { ... })
```